### PR TITLE
Jn ddtr output

### PR DESCRIPTION
**Pull request by jan2nov**
---
name: DDTR output and candidates unscalled output; Python interface update
about: Create a pull request to help us improve

---

**Description of pull request**
Extension of the current python interface. Getting unscaled results from SPD to the user through the python; DDTR writing by dm_range; New introduction of the generic pipeline running through ranges and time_chunks

**Interfaces**
Will this pull request result in a backwards-incompatible interface change: yes

Expected semantic version number increment category (Please indicate x.y.z):y

**Issues this pull request solves**

**New tag of master**

**New deployment**

**Keep branch after merging**
no

**Notes**

### CODE REVIEW

File: include/aa_ddtr_strategy.hpp [hunk 1]
```diff
@@ -94,6 +94,16 @@ namespace astroaccelerate {
     const int* ndms_data() const {
       return m_ndms.data();
     }
+
+    /** \return number of nsamples in the first range */
+    int t_samples() const {
```
Comment (KAdamek) [inline include/aa_ddtr_strategy.hpp]: Rename function to nSamples_in_first_chunk.

File: include/aa_py_filterbank_metadata.hpp [hunk 2]
```diff
@@ -21,6 +21,7 @@ namespace astroaccelerate {
 	int m_nbits;
 	int m_nsamples;
 	int m_nchans;
+	double m_src_raj;
```
Comment (KAdamek) [inline include/aa_py_filterbank_metadata.hpp:24]: I'm not sure what that should be. It would be best to rename it.

File: python/use_python.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 from py_astro_accelerate import *
 
 # Open filterbank file for reading metadata and signal data
-sigproc_input = aa_py_sigproc_input("<input_some_filterbank_data.fil>")
+sigproc_input = aa_py_sigproc_input("/home/novotny/filterbank/ska-mid-b2-small.fil")
```
Comment (KAdamek) [inline python/use_python.py]: Leave old string for fil file

File: python/use_python.py [hunk 4]
```diff
@@ -20,9 +20,9 @@
 # ddtr_plan settings
 # settings: low, high, step, inBin, outBin.
 dm1 = aa_py_dm(0, 370, 0.307, 1, 1)
-dm2 = aa_py_dm(370, 740, 0.652, 2, 2)
-dm3 = aa_py_dm(740, 1480, 1.266, 4, 4)
-dm_list = np.array([dm1, dm2, dm3], dtype=aa_py_dm)
+#dm2 = aa_py_dm(370, 740, 0.652, 2, 2)
+#dm3 = aa_py_dm(740, 1480, 1.266, 4, 4)
```
Comment (KAdamek) [inline python/use_python.py]: Uncomment these

File: src/aa_py_filterbank_metadata.cpp [hunk 5]
```diff
@@ -21,6 +22,7 @@ namespace astroaccelerate {
 	obj->m_nbits = nbits;
 	obj->m_nsamples = nsamples;
 	obj->m_nchans = nchans;
+	obj->m_src_raj = src_raj;
```
Comment (KAdamek) [inline src/aa_py_filterbank_metadata.cpp:25]: Same as before. I do not know what it should represent.

**Review by KAdamek — CHANGES_REQUESTED** [review]
Comments are somewhere.

File: include/aa_pipeline_api.hpp [hunk 6]
```diff
@@ -338,6 +338,22 @@ namespace astroaccelerate {
...
 
+		size_t range(){
+			return m_ddtr_strategy.range();
+		}
+
+		const int* ndms_data(){
+			return m_ddtr_strategy.ndms_data();
+		}
+
+		int tprocessed(){
```
Comment (KAdamek) [inline include/aa_pipeline_api.hpp]: This should be again nSamples_in_first_chunk

File: include/aa_pipeline_api.hpp [hunk 7]
```diff
@@ -338,6 +338,22 @@ namespace astroaccelerate {
 			}
 		}
 
+		size_t range(){
```
Comment (KAdamek) [inline include/aa_pipeline_api.hpp]: This should be get_nRanges

File: include/aa_pipeline_api.hpp [hunk 8]
```diff
@@ -338,6 +338,22 @@ namespace astroaccelerate {
...
+			return m_ddtr_strategy.range();
+		}
+
+		const int* ndms_data(){
```
Comment (KAdamek) [inline include/aa_pipeline_api.hpp]: This should be somethink like get_ndms_array

File: include/aa_ddtr_strategy.hpp (hunk 1 above)
Comment (jan2nov) [reply include/aa_ddtr_strategy.hpp]: The function is not used anymore. Delete the t_samples.

File: python/use_python.py (hunk 3 above)
Comment (jan2nov) [reply python/use_python.py]: Done, also added a comment to the user to add the filterbank file

File: python/use_python.py (hunk 4 above)
Comment (jan2nov) [reply python/use_python.py]: Done

File: include/aa_py_filterbank_metadata.hpp (hunk 2 above)
Comment (jan2nov) [reply include/aa_py_filterbank_metadata.hpp:24]: I believe is the filterbank part. It should be
J2000 Right Ascension
J2000 Declination

File: src/aa_py_filterbank_metadata.cpp (hunk 5 above)
Comment (jan2nov) [reply src/aa_py_filterbank_metadata.cpp:25]: As before mentioned:
m_src_raj = J2000 Right Ascension (hh:mm:ss.ssss)
later if we extend also
m_src_dej = J2000 Declination     (dd:mm:ss.ssss)

File: include/aa_pipeline_api.hpp (hunk 6 above)
Comment (jan2nov) [reply include/aa_pipeline_api.hpp]: Deleted as the function is not used anymore

File: include/aa_pipeline_api.hpp (hunk 7 above)
Comment (jan2nov) [reply include/aa_pipeline_api.hpp]: Renamed as suggested. Changes in all permitted pipelines

File: include/aa_pipeline_api.hpp (hunk 8 above)
Comment (jan2nov) [reply include/aa_pipeline_api.hpp]: Renamed it as suggested

**Pull request merged**